ATParser.cpp

```diff
@@ -155,7 +155,8 @@ int ATParser::vscanf(const char *format, va_list args)
         sscanf(_buffer+offset, _buffer, &count);
 
         // We only succeed if all characters in the response are matched
-        if (count == j) {
+        // and we are at the end of the response
+        if ((count == j) && (format[i-1] == (const char)c)) {
```

Comment:
Ah I see, since whitespace can match "no whitespace". Actually @sarahmarshy's example has a workaround. An "ignored scanset" with whitespace should force scanf to match some whitespace.

I think changing your recv's like this should work:
``` diff
- _parser.recv("+WIND:0:Console active\x0d")
+ _parser.recv("+WIND:0:Console active%*[\x0d]")
```